src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -234,13 +235,94 @@ public AndroidTzData()
+            // We attempt to use tzlookup.xml, which is available on some versions of Android to help
+            // validate non-backward timezone ids
+            // tzlookup.xml is an autogenerated file that contains timezone ids in this form:
+            //
+            // <timezones ianaversion="2019b">
+            //   <countryzones>
+            //     <country code="au" default="Australia/Sydney" everutc="n">
+            //       <id alts="Australia/ACT,Australia/Canberra,Australia/NSW">Australia/Sydney</id>
+            //       ...
+            //       ...
+            //       <id>Australia/Eucla</id>
+            //     </country>
+            //     <country ...>
+            //       ...
+            //       ...
+            //       ...
+            //     </country>
+            //   </countryzones>
+            // </timezones>
+            //
+            // Once the timezone cache is populated with the IDs, we reference tzlookup id tags and
+            // remove IDs and their associated information (byteoffsets and lengths) from the current
+            // AndroidTzData instance.
+            private void FilterBackwardIDs(string lookupPath)
+            {
+                try
+                {
+                    using (StreamReader sr = File.OpenText(lookupPath))
+                    {
+                        var tzLookupIDs = new HashSet<string>();
+                        string? tzLookupLine;
+                        while (sr.Peek() >= 0)
+                        {
+                            if (!(tzLookupLine = sr.ReadLine())!.TrimStart().StartsWith("<id"))
+                                continue;
+
+                            int idStart = tzLookupLine.IndexOf('>') + 1;
+                            int idLength = tzLookupLine.LastIndexOf("</") - idStart;
+                            if (idStart == 0 || idLength < 0)
+                            {
+                                // Either the start tag <id ... > or the end tag </id> are not found
+                                continue;
+                            }
+                            string id = tzLookupLine.Substring(idStart, idLength);
+                            tzLookupIDs.Add(id);
+                        }
+                        var cleanIDs = new List<string>();
+                        var cleanOffsets = new List<int>();
+                        var cleanLengths = new List<int>();
+                        for (int i = 0; i < _ids.GetLength(0); ++i)
+                        {
+                            if (tzLookupIDs.Contains(_ids[i]))
+                            {
+                                cleanIDs.Add(_ids[i]);
+                                cleanOffsets.Add(_byteOffsets[i]);
+                                cleanLengths.Add(_lengths[i]);
+                            }
+                        }
+                        _ids = cleanIDs.ToArray();
```

Comment:
>_ids = cleanIDs.ToArray(); [](http://example.com/codeflow?start=24&length=26)

The question here, if someone try to manually create any of the backward zone (using FindSystemTimeZoneById) will fail. I think it will be better not fail it. 

Can we mark which Ids are backward and filter those only while enumerating the zones?

Reply:
Good point, so instead of removing the backward timezones, we can instead create a parallel array of bools to track which are backward (where that array is updated by this function)?

Reply:
Done, now there is an _isBackwards boolean array that is created and set when the id's and related byteoffset and lengths information are set as well.